main.go

```diff
@@ -92,6 +97,12 @@ func main() {
 		go awsChecker.StartChecking()
 	}
 
-	http.Handle(prometheusPath, promhttp.Handler())
+	handler := promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{})
+
+	if !prometheusExporterMetricsDisabled {
+		handler = promhttp.InstrumentMetricHandler(prometheus.DefaultRegisterer, handler)
```

Comment:
this seems to only add some simple metrics for http calls to the `/metrics` endpoint?

Reply:
Correct, this is what does the `promhttp_` metrics by default. I took inspiration from https://github.com/prometheus/node_exporter/pull/1148 

I was reading through https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promhttp#HandlerFor to how to implement.

The code calls it out in a comment too, here: https://github.com/prometheus/client_golang/blob/v1.12.1/prometheus/promhttp/http.go#L82-L85